### ISSUE TITLE
feat(lessons): Phase 2 option A — cost-only forecast threshold gate (curve-dependent)

### ISSUE THREAD

**Issue by szhygulin** [szhygulin/vaultpilot-dev-framework#199]
## Phase 2 follow-up to [PR #190](https://github.com/szhygulin/vaultpilot-development-agents/pull/190) and [PR #193](https://github.com/szhygulin/vaultpilot-development-agents/pull/193)

Option A from the cost/benefit menu — a hard refusal when the predicted accuracy degradation from adding a lesson exceeds a numerical threshold:

```
predicted_quality_loss = accuracyDegradationFactor(currentBytes + addedBytes)
                       - accuracyDegradationFactor(currentBytes)
if predicted_quality_loss > X: refuse
```

Independent of the LLM's `predictedUtility` (PR #193 option B) — A treats the cost as enough to refuse on its own, B requires utility justification. Both can stack.

## Why deferred

Curve-dependent. Leg-1 of #179 fits at p=0.097 (linear-log); the cost-curve numbers are believable but not statistically significant. A hard numerical gate based on a not-yet-significant curve commits to thresholds that may shift when leg-2 data lands.

Per the project CLAUDE.md "Pre-dispatch bake-window check" rule, this issue should not be dispatched until [#179](https://github.com/szhygulin/vaultpilot-development-agents/issues/179)'s leg 2 (K=13 combined dataset) brings the accuracy fit to p < 0.05 with a comparable slope. Empirically:
- p < 0.05 → curve is defensible as a basis for a hard gate
- threshold X tuned from the K=13 distribution of `(currentBytes, predictedUtility, costScore)` triples B (PR #193) is now collecting at write-time
- correlation of `predictedUtility` vs subsequent `reinforcementRuns.length` informs whether A or B is the better gate

## Approach

Add to `runIssueCore.ts maybeAppendSummary` after the dedup gate (G), before B's utility gate:

```ts
const projectedFactor = accuracyDegradationFactor(currentBytes + estimatedAdded);
const currentFactor = accuracyDegradationFactor(currentBytes);
const deltaFactor = projectedFactor - currentFactor;
if (deltaFactor > resolveCostForecastThreshold()) {
  return { summarySkipReason: "cost-forecast-exceeds-threshold" };
}
```

Env: `VP_DEV_COST_FORECAST_THRESHOLD` (default TBD by K=13 distribution).

## Out of scope (Phase 3)

Combined gate orchestration (A + B + E coordinated decision); per-agent threshold overrides; gate-decision distribution dashboard.

**Comment by szhygulin** [comment]
The bake-window precondition encoded in this issue's body is not yet satisfied — three independent gate-conditions all fail as of today (2026-05-07).

**The issue's own gate statement** (top of body): _"this issue should not be dispatched until #179's leg 2 (K=13 combined dataset) brings the accuracy fit to p < 0.05 with a comparable slope."_ Status against each clause:

1. **Accuracy fit p < 0.05** — `feature-plans/issue-179-results-phase2.md` (2026-05-06): every one of the 28 cells returned `decision=implement`, so the quality-composite is uniform 0.75 across all 10 sizes. The accuracy regression is **degenerate** (R²=NaN, "no curve to fit"). There is no measured accuracy curve a hard-refusal threshold X could be calibrated against.

2. **Comparable slope** — leg 1 fit at p≈0.097 (linear-log), leg 2's only non-degenerate fit is the **cost-based** quadratic at **F(2,7)=3.13, p=0.107**. Both legs sit on the wrong side of α=0.05; the writeup itself recommends "hold the placeholder" and file a follow-up "Curve study v2 — harder issue set + operator rubrics."

3. **K=13 combined dataset** — leg 2 ran K=10 sizes, not K=13. The combined corpus described in the issue body doesn't yet exist in the form the gate names.

Additionally, the alternative threshold-tuning corpus the issue body anticipates ("threshold X tuned from the K=13 distribution of `(currentBytes, predictedUtility, costScore)` triples B (PR #193) is now collecting") has had **~24h of accumulation** (PR #193 merged 2026-05-06T13:44Z → 2026-05-07 dispatch). One day of `predictedUtility`/`costScore` triples is far too sparse to fit a defensible numerical threshold against.

Net: every signal a hard cost-only refusal gate would commit to a number against is either degenerate, not significant at α=0.05, or pre-bake. Per the project CLAUDE.md "Pre-dispatch bake-window check" rule, this is the failure mode the deferral was written to prevent — committing thresholds against guessed/insignificant curves rather than observed ones.

Three alternatives, in order of my preference:

1. **Defer to the stated horizon** (do nothing in code). Close as blocked/dependent and re-open when (a) a curve-study v2 with harder issues + operator pushback-accuracy rubrics yields an accuracy fit at p<0.05, AND (b) PR #193's `(currentBytes, predictedUtility, costScore)` corpus has accumulated long enough to characterize the joint distribution at K≥13. The leg-2 writeup already names the v2 study as the natural successor.

2. **Reframe as a plumbing-only Phase B0 with a `STUDY_PENDING` placeholder.** Ship just the structural piece: a `resolveCostForecastThreshold()` helper that reads `VP_DEV_COST_FORECAST_THRESHOLD` and **defaults to `Infinity` (gate disabled)**, the corresponding `maybeAppendSummary` call-site that emits `summarySkipReason: "cost-forecast-exceeds-threshold"` when the threshold trips, and a unit test asserting "with default threshold, no skip ever fires." No behavioral change at default, no calibrated number committed — the wiring is there for the day the curve becomes significant. Mirrors the `STUDY_PENDING` pattern already shipped for #179's `contextCostCurve.ts`.

3. **Synthesize the corpus first.** File a v2 curve-study issue (harder issues, operator pushback-accuracy rubrics, target K≥13) and only dispatch #199 against the resulting fit. The leg-2 writeup explicitly recommends this as the natural follow-up. Cost: another ~$140-ish for the study, plus operator scoring time.

Which path do you want? My instinct is option (1) — neither (2) nor (3) lands the actual gate, and (2) without a calibrated number is just decorative scaffolding that may need to be re-shaped once the curve does land (e.g., the function signature might want `(currentBytes, addedBytes, predictedUtility)` rather than the current `(currentBytes, addedBytes)` if the v2 fit shows the joint with B-utility is the better predictor).

— Raman (agent-5acc)

**Comment by szhygulin** [comment]
experiment result were shipped. proceed according to the plan
